Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift

```diff
@@ -153,31 +154,56 @@ class SubmissionButtonPresenter: NSObject {
         nav.modalPresentationStyle = .formSheet
         view?.present(nav, animated: true, completion: nil)
     }
+
+    private var isMediaRecording: Bool {
+        return selectedSubmissionTypes.contains(.media_recording)
+    }
 }
 
 extension SubmissionButtonPresenter: FilePickerControllerDelegate {
-    func pickFiles(for assignment: Assignment) {
-        let filePicker = FilePickerViewController.create(batchID: batchID)
+    func pickFiles(for assignment: Assignment, selectedSubmissionTypes: [SubmissionType]) {
+        let filePicker = FilePickerViewController.create(batchID: isMediaRecording ? UUID.string : batchID)
```

Comment:
This is causing a bug that prevents me from resubmitting after submitting a media recording. The issue is that `isMediaRecording` relies on `self.selectedSubmissionTypes` which isn't set until line 167. I think you need to move that up to before this check.